### PR TITLE
[Done] SavedPosts API Endpoints

### PR DESCRIPTION
**Pull request by parthsujalshah**
added serializers and views for saved posts

### CODE REVIEW

**Comment by jsparmani** [comment]
@parthsujalshah pre-commit tests are failing, please fix it.

File: buying_selling/users/api/views.py [hunk 1]
```diff
@@ -70,3 +72,43 @@ def perform_update(self, serializer):
...
+    def perform_create(self, serializer):
+        payload = jwt_decoder(self.request.headers['Authorization'].split()[1])
+        serializer.save(author_id=payload['user_id'])
+
+    def list(self, request, *args, **kwargs):
+        payload = jwt_decoder(self.request.headers['Authorization'].split()[1])
+        current_user = MyUser.objects.get(id=payload['user_id'])
+        queryset = self.filter_queryset(SavedPosts.objects.filter(author=current_user))
+        page = self.paginate_queryset(queryset)
+        if page not in None:
```
Comment (Saurav-Shrivastav) [inline buying_selling/users/api/views.py:109]: Iterating over a NoneType object isn't possible. Is there anything else that you wanted to achieve?

**Comment by jsparmani** [comment]
So, @parthsujalshah @Saurav-Shrivastav what's the status, I was out for a couple of days so I don't have any checks on it. Is it ready to be merged?

**Comment by jsparmani** [comment]
@parthsujalshah I am getting error on trying to hit POST on the route. It's some error saying _list index out of range_ . Can u check that out.

**Pull request merged**